pints/_abc/_abc_smc.py

```diff
@@ -0,0 +1,176 @@
+#
+# ABC SMC method
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+
+
+class ABCSMC(pints.ABCSampler):
+    """
+    ABC-SMC Algorithm  See, for example, [1]_. In each iteration of the
+    algorithm, the following steps occur::
+
+        theta* ~ p_(t-1)(theta), i.e. sample parameters from previous
+            intermediate distribution
+        theta** ~ K(theta|theta*), i.e. perturb theta* to obtain to new point
+        x ~ p(x|theta**), i.e. sample data from sampling distribution
+        if s(x) < threshold_(t), theta* added to list of samples[t]
+
+    After we have obtained nr_samples samples, t is advanced, and weights
```

Comment:
nr_samples has not been introduced. Will it be used again? If it's code it should be ``nr_samples``